### PR TITLE
ARROW-7518: [Python] Use PYARROW_WITH_HDFS when building wheels, conda packages

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
@ursabot crossbow submit -g wheel -g conda

File: dev/tasks/conda-recipes/arrow-cpp/bld.bat [hunk 1]
```diff
@@ -21,6 +21,7 @@ cmake -G "Ninja" ^
       -DARROW_PYTHON:BOOL=ON ^
       -DARROW_MIMALLOC:BOOL=ON ^
       -DARROW_FLIGHT:BOOL=ON ^
+      -DARROW_HDFS:BOOL=ON ^
```
Comment (wesm) [inline dev/tasks/conda-recipes/arrow-cpp/bld.bat]: Note this isn't needed

https://github.com/apache/arrow/blob/master/cpp/CMakeLists.txt#L289

Is `pyarrow._hdfs` optional now?

Comment (kszucs) [reply dev/tasks/conda-recipes/arrow-cpp/bld.bat]: We have two filesystem implementations, the old python one and the new C++ implementation. `_hdfs` and `_s3` are the bindings for the C++ implementation. `pyarrow.hdfs` is not optional but `pyarrow._hdfs` (exposed under `pyarrow.fs`) is optional.

**Comment by kszucs** [comment]
@ursabot crossbow submit -g wheel -g conda

**Comment by kszucs** [comment]
@ursabot crossbow submit -g wheel -g conda

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot crossbow submit conda-linux-gcc-py27

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#87410)](https://ci.ursalabs.org/#builders/98/builds/445) builder failed.

Revision: 50ad951c897230c85933ef92085fed12decc0083

Crossbow: `'python crossbow.py ...'` step's stderr:
```
Traceback (most recent call last):
  File "crossbow.py", line 300, in push
    self.origin.push(self._updated_refs, callbacks=callbacks)
  File "/opt/conda/lib/python3.7/site-packages/pygit2/remote.py", line 485, in push
    check_error(err)
  File "/opt/conda/lib/python3.7/site-packages/pygit2/errors.py", line 64, in check_error
    raise GitError(message)
_pygit2.GitError: unexpected HTTP status code: 403

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "crossbow.py", line 1305, in <module>
    crossbow(obj={}, auto_envvar_prefix='CROSSBOW')
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "crossbow.py", line 1166, in submit
    queue.push()
  File "crossbow.py", line 304, in push
    .format(self._updated_refs))
RuntimeError: Failed to push updated references, potentially because of credential issues: ['refs/heads/ursabot-442-azure-conda-linux-gcc-py27', 'refs/tags/ursabot-442-azure-conda-linux-gcc-py27', 'refs/heads/ursabot-442']
```

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#87411)](https://ci.ursalabs.org/#builders/98/builds/446) builder failed.

Revision: 50ad951c897230c85933ef92085fed12decc0083

Crossbow: `'python crossbow.py ...'` step's stderr:
```
Traceback (most recent call last):
  File "crossbow.py", line 300, in push
    self.origin.push(self._updated_refs, callbacks=callbacks)
  File "/opt/conda/lib/python3.7/site-packages/pygit2/remote.py", line 485, in push
    check_error(err)
  File "/opt/conda/lib/python3.7/site-packages/pygit2/errors.py", line 64, in check_error
    raise GitError(message)
_pygit2.GitError: unexpected HTTP status code: 403

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "crossbow.py", line 1305, in <module>
    crossbow(obj={}, auto_envvar_prefix='CROSSBOW')
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "crossbow.py", line 1166, in submit
    queue.push()
  File "crossbow.py", line 304, in push
    .format(self._updated_refs))
RuntimeError: Failed to push updated references, potentially because of credential issues: ['refs/heads/ursabot-442-azure-conda-linux-gcc-py27', 'refs/tags/ursabot-442-azure-conda-linux-gcc-py27', 'refs/heads/ursabot-442']
```

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#87412)](https://ci.ursalabs.org/#builders/98/builds/447) builder failed.

Revision: 50ad951c897230c85933ef92085fed12decc0083

Crossbow: `'python crossbow.py ...'` step's stderr:
```
Traceback (most recent call last):
  File "crossbow.py", line 300, in push
    self.origin.push(self._updated_refs, callbacks=callbacks)
  File "/opt/conda/lib/python3.7/site-packages/pygit2/remote.py", line 485, in push
    check_error(err)
  File "/opt/conda/lib/python3.7/site-packages/pygit2/errors.py", line 64, in check_error
    raise GitError(message)
_pygit2.GitError: unexpected HTTP status code: 403

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "crossbow.py", line 1305, in <module>
    crossbow(obj={}, auto_envvar_prefix='CROSSBOW')
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "crossbow.py", line 1166, in submit
    queue.push()
  File "crossbow.py", line 304, in push
    .format(self._updated_refs))
RuntimeError: Failed to push updated references, potentially because of credential issues: ['refs/heads/ursabot-442-azure-conda-linux-gcc-py27', 'refs/tags/ursabot-442-azure-conda-linux-gcc-py27', 'refs/heads/ursabot-442']
```

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#87415)](https://ci.ursalabs.org/#builders/98/builds/448) builder has been succeeded.

Revision: 50ad951c897230c85933ef92085fed12decc0083

Submitted crossbow builds: [ursa-labs/crossbow @ ursabot-442](https://github.com/ursa-labs/crossbow/branches/all?query=ursabot-442)

|Task|Status|
|----|------|
|conda-linux-gcc-py27|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-442-azure-conda-linux-gcc-py27)|

**Comment by kszucs** [comment]
@ursabot crossbow submit -g wheel -g conda

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#87416)](https://ci.ursalabs.org/#builders/98/builds/449) builder has been succeeded.

Revision: 50ad951c897230c85933ef92085fed12decc0083

Submitted crossbow builds: [ursa-labs/crossbow @ ursabot-443](https://github.com/ursa-labs/crossbow/branches/all?query=ursabot-443)

|Task|Status|
|----|------|
|conda-linux-gcc-py27|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-linux-gcc-py27)|
|conda-linux-gcc-py36|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-linux-gcc-py36)|
|conda-linux-gcc-py37|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-linux-gcc-py37)|
|conda-linux-gcc-py38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-linux-gcc-py38)|
|conda-osx-clang-py27|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-osx-clang-py27)|
|conda-osx-clang-py36|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-osx-clang-py36)|
|conda-osx-clang-py37|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-osx-clang-py37)|
|conda-osx-clang-py38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-osx-clang-py38)|
|conda-win-vs2015-py36|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-win-vs2015-py36)|
|conda-win-vs2015-py37|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-win-vs2015-py37)|
|conda-win-vs2015-py38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-conda-win-vs2015-py38)|
|wheel-manylinux1-cp27m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux1-cp27m)|
|wheel-manylinux1-cp27mu|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux1-cp27mu)|
|wheel-manylinux1-cp35m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux1-cp35m)|
|wheel-manylinux1-cp36m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux1-cp36m)|
|wheel-manylinux1-cp37m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux1-cp37m)|
|wheel-manylinux1-cp38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux1-cp38)|
|wheel-manylinux2010-cp27m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2010-cp27m)|
|wheel-manylinux2010-cp27mu|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2010-cp27mu)|
|wheel-manylinux2010-cp35m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2010-cp35m)|
|wheel-manylinux2010-cp36m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2010-cp36m)|
|wheel-manylinux2010-cp37m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2010-cp37m)|
|wheel-manylinux2010-cp38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2010-cp38)|
|wheel-manylinux2014-cp35m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2014-cp35m)|
|wheel-manylinux2014-cp36m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2014-cp36m)|
|wheel-manylinux2014-cp37m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2014-cp37m)|
|wheel-manylinux2014-cp38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-443-azure-wheel-manylinux2014-cp38)|
|wheel-osx-cp27m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp35m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp36m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp37m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp38|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-win-cp36m|[Appveyor](https://ci.appveyor.com/project/ursa-labs/crossbow/history)|
|wheel-win-cp37m|[Appveyor](https://ci.appveyor.com/project/ursa-labs/crossbow/history)|
|wheel-win-cp38|[Appveyor](https://ci.appveyor.com/project/ursa-labs/crossbow/history)|

File: python/manylinux201x/build_arrow.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+  #!/bin/bash
```
Comment (nealrichardson) [inline python/manylinux201x/build_arrow.sh]:
```suggestion
#!/bin/bash
```
This looks like an accident

Comment (kszucs) [reply python/manylinux201x/build_arrow.sh]: It is, thanks for spotting that!

**Comment by kszucs** [comment]
@ursabot crossbow submit -g wheel

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#87615)](https://ci.ursalabs.org/#builders/98/builds/454) builder has been succeeded.

Revision: 77311b8bea246560478145eb112f27662f9eb45c

Submitted crossbow builds: [ursa-labs/crossbow @ ursabot-446](https://github.com/ursa-labs/crossbow/branches/all?query=ursabot-446)

|Task|Status|
|----|------|
|wheel-manylinux1-cp27m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux1-cp27m)|
|wheel-manylinux1-cp27mu|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux1-cp27mu)|
|wheel-manylinux1-cp35m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux1-cp35m)|
|wheel-manylinux1-cp36m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux1-cp36m)|
|wheel-manylinux1-cp37m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux1-cp37m)|
|wheel-manylinux1-cp38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux1-cp38)|
|wheel-manylinux2010-cp27m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2010-cp27m)|
|wheel-manylinux2010-cp27mu|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2010-cp27mu)|
|wheel-manylinux2010-cp35m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2010-cp35m)|
|wheel-manylinux2010-cp36m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2010-cp36m)|
|wheel-manylinux2010-cp37m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2010-cp37m)|
|wheel-manylinux2010-cp38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2010-cp38)|
|wheel-manylinux2014-cp35m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2014-cp35m)|
|wheel-manylinux2014-cp36m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2014-cp36m)|
|wheel-manylinux2014-cp37m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2014-cp37m)|
|wheel-manylinux2014-cp38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-446-azure-wheel-manylinux2014-cp38)|
|wheel-osx-cp27m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp35m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp36m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp37m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp38|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-win-cp36m|[Appveyor](https://ci.appveyor.com/project/ursa-labs/crossbow/history)|
|wheel-win-cp37m|[Appveyor](https://ci.appveyor.com/project/ursa-labs/crossbow/history)|
|wheel-win-cp38|[Appveyor](https://ci.appveyor.com/project/ursa-labs/crossbow/history)|

**Comment by kszucs** [comment]
The windows errors should be resolved by https://github.com/apache/arrow/pull/6242

**Comment by kszucs** [comment]
@ursabot crossbow submit -g wheel -g conda

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#87965)](https://ci.ursalabs.org/#builders/98/builds/459) builder has been succeeded.

Revision: 6dd82a6d3443d678639af8ee78233e83fd53b5eb

Submitted crossbow builds: [ursa-labs/crossbow @ ursabot-450](https://github.com/ursa-labs/crossbow/branches/all?query=ursabot-450)

|Task|Status|
|----|------|
|conda-linux-gcc-py27|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-linux-gcc-py27)|
|conda-linux-gcc-py36|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-linux-gcc-py36)|
|conda-linux-gcc-py37|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-linux-gcc-py37)|
|conda-linux-gcc-py38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-linux-gcc-py38)|
|conda-osx-clang-py27|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-osx-clang-py27)|
|conda-osx-clang-py36|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-osx-clang-py36)|
|conda-osx-clang-py37|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-osx-clang-py37)|
|conda-osx-clang-py38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-osx-clang-py38)|
|conda-win-vs2015-py36|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-win-vs2015-py36)|
|conda-win-vs2015-py37|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-win-vs2015-py37)|
|conda-win-vs2015-py38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-conda-win-vs2015-py38)|
|wheel-manylinux1-cp27m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux1-cp27m)|
|wheel-manylinux1-cp27mu|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux1-cp27mu)|
|wheel-manylinux1-cp35m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux1-cp35m)|
|wheel-manylinux1-cp36m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux1-cp36m)|
|wheel-manylinux1-cp37m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux1-cp37m)|
|wheel-manylinux1-cp38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux1-cp38)|
|wheel-manylinux2010-cp27m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2010-cp27m)|
|wheel-manylinux2010-cp27mu|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2010-cp27mu)|
|wheel-manylinux2010-cp35m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2010-cp35m)|
|wheel-manylinux2010-cp36m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2010-cp36m)|
|wheel-manylinux2010-cp37m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2010-cp37m)|
|wheel-manylinux2010-cp38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2010-cp38)|
|wheel-manylinux2014-cp35m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2014-cp35m)|
|wheel-manylinux2014-cp36m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2014-cp36m)|
|wheel-manylinux2014-cp37m|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2014-cp37m)|
|wheel-manylinux2014-cp38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-450-azure-wheel-manylinux2014-cp38)|
|wheel-osx-cp27m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp35m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp36m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp37m|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-osx-cp38|[TravisCI](https://travis-ci.org/ursa-labs/crossbow/branches)|
|wheel-win-cp36m|[Appveyor](https://ci.appveyor.com/project/ursa-labs/crossbow/history)|
|wheel-win-cp37m|[Appveyor](https://ci.appveyor.com/project/ursa-labs/crossbow/history)|
|wheel-win-cp38|[Appveyor](https://ci.appveyor.com/project/ursa-labs/crossbow/history)|

**Comment by kszucs** [comment]
The coda-win failures should be resolved by https://github.com/apache/arrow/pull/6242
And we should deal with the OSX-wheel failures in a follow-up PR.

File: python/manylinux201x/build_arrow.sh [hunk 3]
```diff
@@ -98,38 +98,40 @@ echo "=== (${PYTHON_VERSION}) Building Arrow C++ libraries ==="
...
-    -DZLIB_ROOT=/usr/local \
-    -DCMAKE_INSTALL_PREFIX=/arrow-dist \
-    -DCMAKE_INSTALL_LIBDIR=lib \
-    -DARROW_BUILD_TESTS=OFF \
+PATH="${CPYTHON_PATH}/bin:${PATH}" cmake \
+    -DARROW_BOOST_USE_SHARED=ON \
     -DARROW_BUILD_SHARED=ON \
     -DARROW_BUILD_STATIC=OFF \
-    -DARROW_BOOST_USE_SHARED=ON \
+    -DARROW_BUILD_TESTS=OFF \
```
Comment (kszucs) [inline python/manylinux201x/build_arrow.sh]: Just sorted the options alphabetically.

File: dev/tasks/conda-recipes/pyarrow/meta.yaml [hunk 4]
```diff
@@ -44,13 +44,14 @@ requirements:
 test:
   imports:
     - pyarrow
-    - pyarrow.fs
-    - pyarrow.dataset  # [not py==27]
```
Comment (kou) [inline dev/tasks/conda-recipes/pyarrow/meta.yaml]: Is it expected?

File: dev/tasks/conda-recipes/arrow-cpp/bld.bat (hunk 1 above)
Comment (kou) [reply dev/tasks/conda-recipes/arrow-cpp/bld.bat]: `-DARROW_PYTHON:BOOL=ON` enables `ARROW_HDFS` automatically. So we don't need to specify `-DARROW_HDFS:BOOL=ON` explicitly.

**Review by kou — APPROVED** [review]
+1

File: dev/tasks/conda-recipes/pyarrow/meta.yaml (hunk 4 above)
Comment (kszucs) [reply dev/tasks/conda-recipes/pyarrow/meta.yaml]: Nice catch, rebase issue.

**Comment by kszucs** [comment]
@ursabot crossbow submit -g conda

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#88151)](https://ci.ursalabs.org/#builders/98/builds/465) builder has been succeeded.

Revision: 1e3f091954afb1ebf1ad0ba789c84bda8134ff29

Submitted crossbow builds: [ursa-labs/crossbow @ ursabot-455](https://github.com/ursa-labs/crossbow/branches/all?query=ursabot-455)

|Task|Status|
|----|------|
|conda-linux-gcc-py27|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-linux-gcc-py27)|
|conda-linux-gcc-py36|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-linux-gcc-py36)|
|conda-linux-gcc-py37|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-linux-gcc-py37)|
|conda-linux-gcc-py38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-linux-gcc-py38)|
|conda-osx-clang-py27|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-osx-clang-py27)|
|conda-osx-clang-py36|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-osx-clang-py36)|
|conda-osx-clang-py37|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-osx-clang-py37)|
|conda-osx-clang-py38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-osx-clang-py38)|
|conda-win-vs2015-py36|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-win-vs2015-py36)|
|conda-win-vs2015-py37|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-win-vs2015-py37)|
|conda-win-vs2015-py38|[Azure](https://dev.azure.com/ursa-labs/crossbow/_build/latest?definitionId=1&branchName=ursabot-455-azure-conda-win-vs2015-py38)|

File: dev/tasks/conda-recipes/arrow-cpp/bld.bat (hunk 1 above)
Comment (kszucs) [reply dev/tasks/conda-recipes/arrow-cpp/bld.bat]: I prefer to be explicit about these options.

File: python/manylinux1/build_arrow.sh [hunk 5]
```diff
@@ -97,36 +97,38 @@ echo "=== (${PYTHON_VERSION}) Building Arrow C++ libraries ==="
...
-cmake -DCMAKE_BUILD_TYPE=Release \
-    -DARROW_DEPENDENCY_SOURCE="SYSTEM" \
-    -DCMAKE_INSTALL_PREFIX=/arrow-dist \
-    -DCMAKE_INSTALL_LIBDIR=lib \
-    -DARROW_BUILD_TESTS=OFF \
-    -DARROW_BUILD_SHARED=ON \
+cmake \
+    -DCMAKE_BUILD_TYPE=Release \
     -DARROW_BOOST_USE_SHARED=ON \
+    -DARROW_BUILD_SHARED=ON \
```
Comment (pitrou) [inline python/manylinux1/build_arrow.sh]: Not in the scope for this PR, but should we add `ARROW_BUILD_STATIC=OFF` here?

**Review by pitrou — APPROVED** [review]
+1, thanks for doing this

**Review by nealrichardson — APPROVED** [review]
+1, merging now

File: python/manylinux1/build_arrow.sh (hunk 5 above)
Comment (kszucs) [reply python/manylinux1/build_arrow.sh]: Probably, created jira https://issues.apache.org/jira/browse/ARROW-7676

**Pull request closed without merging**